src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1228,6 +1228,11 @@ private boolean populateRecommendation(Map.Entry<String, Terms> termEntry,
             }
         }
 
+        //Set all existing notifications
+        for (RecommendationNotification recommendationNotification : notifications) {
+            recommendationModel.addNotification(recommendationNotification);
+        }
+
```

Comment:
Is this the right place to add these notifications? Can you please paste an example JSON object that has this notification

Reply:
The reason notifications aren't being set is because the function checks if the flag isRecommendedCPURequestAvailable is set to false. In such cases, these notifications are lost, despite the existing code intending to set a notification here.


https://privatebin.corp.redhat.com/?0252306990b9c2fc#5KA1424pqZhQbbogX7uD8ho36VJk5jngts69uU9F4jYV